pkg/api/v1/defaults.go

```diff
@@ -25,232 +25,255 @@ import (
 
 func addDefaultingFuncs(scheme *runtime.Scheme) {
 	scheme.AddDefaultingFuncs(
-		func(obj *PodExecOptions) {
-			obj.Stdout = true
-			obj.Stderr = true
-		},
-		func(obj *PodAttachOptions) {
-			obj.Stdout = true
-			obj.Stderr = true
-		},
-		func(obj *ReplicationController) {
-			var labels map[string]string
-			if obj.Spec.Template != nil {
-				labels = obj.Spec.Template.Labels
-			}
-			// TODO: support templates defined elsewhere when we support them in the API
-			if labels != nil {
-				if len(obj.Spec.Selector) == 0 {
-					obj.Spec.Selector = labels
-				}
-				if len(obj.Labels) == 0 {
-					obj.Labels = labels
-				}
-			}
-			if obj.Spec.Replicas == nil {
-				obj.Spec.Replicas = new(int32)
-				*obj.Spec.Replicas = 1
-			}
-		},
-		func(obj *Volume) {
-			if util.AllPtrFieldsNil(&obj.VolumeSource) {
-				obj.VolumeSource = VolumeSource{
-					EmptyDir: &EmptyDirVolumeSource{},
-				}
-			}
-		},
-		func(obj *ContainerPort) {
-			if obj.Protocol == "" {
-				obj.Protocol = ProtocolTCP
-			}
-		},
-		func(obj *Container) {
-			if obj.ImagePullPolicy == "" {
-				// Ignore error and assume it has been validated elsewhere
-				_, tag, _ := parsers.ParseImageName(obj.Image)
+		SetDefaults_PodExecOptions,
```

Comment:
@smarterclayton - maybe for consistency we should have the names of those functions in exactly the same way as we have for conversions and deep copies. So that would mean:
SetDefaults_v1_PodExectOptions

?